src/bentoml_cli/env_manager.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+import re
+import sys
+import typing as t
+import logging
+import functools
+
+import click
+
+from bentoml.exceptions import NotFound
+
+from bentoml._internal.env_manager import EnvManager
+
+logger = logging.getLogger(__name__)
+
+
+def remove_env_arg(cmd_args: list[str]) -> list[str]:
+    """
+    Removes `--env <env_name>` or `--env=<env_name>` from sys.argv
+    """
+    indices_to_remove: list[int] = []
+    for i, arg in enumerate(cmd_args):
+        # regex matching click.option
+        if re.match(r"^--env=?.*", arg):
+            indices_to_remove.append(i)
+            if "=" not in arg:
+                indices_to_remove.append(i + 1)
+
+    new_cmd_args: list[str] = []
+    for i, arg in enumerate(cmd_args):
+        if i not in indices_to_remove:
+            new_cmd_args.append(arg)
+    return new_cmd_args
+
+
+def env_manager(func):
+    from bentoml._internal.configuration.containers import BentoMLContainer
+
+    bento_store = BentoMLContainer.bento_store.get()
+
+    @click.option(
+        "--env",
+        type=click.Choice(["conda"]),
+        default=None,
+        help="Environment to run the command in",
+        show_default=True,
+    )
+    @functools.wraps(func)
+    def wrapper(*args: t.Any, **kwargs: t.Any) -> t.Any:
+        env = kwargs.pop("env")
+        if env is not None:
+            print("loading env_manager")
+            bento_tag = kwargs["bento"]
+
+            # if bento_tag is a bento in the bento_store, use a persistent env
+            try:
+                bento = bento_store.get(bento_tag)
+                from_bento_store = True
+                bento_path = bento._fs.getsyspath("")
+            except NotFound:
+                # env created will be ephemeral
+                from_bento_store = False
+            bento_env = EnvManager.get_environment(
+                env_name=bento_tag,
+                env_type=env,
+                from_bento_store=from_bento_store,
+                bento_path=bento_path,
```

Comment:
```suggestion
                bento_path=bento.path or "",
```
There is a property `path` for bento which is the equivalent of this.

If you need to get other path then you can do
```python
path = bento.path_of("/")
```

Reply:
ahh, thanks for pointing that out